cli/azd/cmd/container.go

```diff
@@ -162,21 +162,14 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		return writer
 	})
 
-	container.RegisterSingleton(func() flagsWithEnv {
-		// Get the current cmd flags for the executing command
-		var currentFlags flags
-		err := container.Resolve(&currentFlags)
+	container.RegisterSingleton(func(cmd *cobra.Command) envFlag {
+		envValue, err := cmd.Flags().GetString(environmentNameFlag)
 		if err != nil {
-			return &envFlag{}
+			// This is probably an error
+			return envFlag{}
```

Comment:
This is probably an error, but I cannot validate this without running all `azd` commands, so we should revisit this post-release. This is where I'm a little worried about runtime registration...

Reply:
This is great - I wasn't aware  you could retrieve the flag value without getting an instance of the pointer it was bound to.  This greatly simplifies what I was attempting to do with the `flagsWithEnv`.